src/operators.jl

```diff
@@ -292,39 +348,201 @@ function resolve_linear_term(expr, constr_id::Int, m::PODNonlinearModel)
+        var_idxs = []
+        scalar = 1.0
+        for i in 1:length(expr.args)
+            if isa(expr.args[i], Float64) || isa(expr.args[i], Int)
+                scalar *= expr.args[i]
+                continue
+            end
+            (isa(expr.args[i], Symbol)) && continue
+            (expr.args[i].head == :ref) && isa(expr.args[i].args[2], Int) && push!(var_idxs, expr.args[i].args[2])
+            !isempty(var_idxs) && m.var_type_lifted[var_idxs[end]] != :Bin && return false, expr
+            if (expr.args[i].head == :call)
+                down_check, linear_lift_var = resolve_linear_term(expr.args[i], constr_id, m)
+                !down_check && return false, expr
+                m.var_type_lifted[linear_lift_var.args[2]] != :Bin && false, expr
+                push!(var_idxs, linear_lift_var.args[2])
+                continue
+            end
+        end
+        if length(var_idxs) >= 2
+            term_key = [Expr(:ref, :x, idx) for idx in var_idxs]
+            if term_key in keys(m.nonlinear_terms)
+                return true, lift_nl_term(m, term_key, constr_id, scalar)
+            else
+                store_nl_term(m, term_key, var_idxs, :binprod, :*, binprod)
+                return true, lift_nl_term(m, term_key, constr_id, scalar)
+            end
+        end
+    elseif (expr.args[1] == :^) && length(expr.args) == 3
+        # Pattern: (x)^(>2), where x is binary variable
+        var_idxs = []
+        power_scalar = 0
+        scalar = 1.0
+        for i in 2:length(expr.args)
+            if isa(expr.args[i], Float64) || isa(expr.args[i], Int)
+                power_scalar += expr.args[i]
+                continue
+            end
+            (isa(expr.args[i], Symbol)) && continue
+            (expr.args[i].head == :ref) && isa(expr.args[i].args[2], Int) && push!(var_idxs, expr.args[i].args[2])
+            !isempty(var_idxs) && m.var_type_lifted[var_idxs[end]] != :Bin && return false, expr
+            if (expr.args[i].head == :call)
+                down_check, linear_lift_var = resolve_linear_term(expr.args[i], constr_id, m)
+                !down_check && return false, expr
+                m.var_type_lifted[linear_lift_var.args[2]] != :Bin && false, expr
+                push!(var_idxs, linear_lift_var.args[2])
+                continue
+            end
+        end
+        if length(var_idxs) == 1 && power_scalar > 2.0
+            term_key = [Expr(:ref, :x, var_idxs[1]) for i in 1:power_scalar]
+            if term_key in keys(m.nonlinear_terms)
+                return true, lift_nl_term(m, term_key, constr_id, scalar)
+            else
+                store_nl_term(m, term_key, var_idxs, :binprod, :*, binprod)
+                return true, lift_nl_term(m, term_key, constr_id, scalar)
+            end
+        end
+    end
+
+    return false, expr
+end
+
 """
+    Recognize prodcuts of binary variables and multilinear products
```

Comment:
Does this function recognize a mixed product. 
For instance something like x_1 x_2 z_1 z_2. 
What is done for a mixed product?

Reply:
Yes. Take your example, it will take x_1*x_2*z_1*z_2 (by default) as a (cont) * (bin), where (cont)->(bilinear term) and (bin)->(binprod).